selfdrive/thermald/fan_controller.py

```diff
@@ -93,10 +94,10 @@ def update(self, max_cpu_temp: float, ignition: bool) -> int:
       self.controller.reset()
 
     fan_pwr_out = -int(self.controller.update(
-                      setpoint=75,
-                      measurement=max_cpu_temp,
-                      feedforward=interp(max_cpu_temp, [60.0, 100.0], [0, -80])
-                    ))
+                                        setpoint=self.target,
+                                        measurement=max_cpu_temp,
+                                        feedforward=interp(self.target, [60, 100], [-80, 0])
```

Comment:
This interp is wrong. To make it match the old controller the result would need to be 30.

We thought about it a bit more, and changing it to an actual PID doesn't make it easier to reason about. It would imply that it's possible to change the setpoint without retuning the values, which is probably not true. If you do that you would need to find the relationship between kp and ff that works at all setpoints.

Since we don't plan on doing that anytime soon we'll just leave the controller as is.

Reply:
I need to push back on the thought that it's not any easier to reason about. 
The current function in feedforward is a second P-only controller with a setpoint of 60. it provides no FF information.

it is actually helpful to separate the controller behavior and highlight both of the current misunderstandings in this comment: 
> The purpose of the feedforward on temperature is due to the fact that the fan speed should not go to 0 when the error becomes 0. In fact, it's used to describe a decent estimate of the steady-state relationship between temp and fan speed.
> 
> If you read the implementation of the PIController, you'll see :)

This is both not the purpose of feedforward, and the function here is inverse of the relationship that we want to use in FF. 

it would be better to use the PID class correctly so that when reasoning about FF we are actually reasoning about FF. 
I set "feedforward=30" on a new commit. This matches the current behavior. 

Changing the setpoint would be different between the current and new code however the FF increasing with increased setpoint temp would be incorrect and is what would occur currently. 
it would be beneficial to model FF in the form interp(self.target, [ssTemp(fan_max), ssTemp(fan_min)], [-fan_max, fan_min]) 
However, it is not beneficial to do this unless/untill an adaptive setpoint would be desired. but as you pointed out a value of 30 is the current implementation.

Reply:
>It would imply that it's possible to change the setpoint without retuning the values, which is probably not true. 

this is actually true. The controller and FF tuning could be improved but the setpoint does not effect the FF function or the p-gain.